File: android/src/main/kotlin/com/example/amps_sdk/data/InstanceChannelHelper.kt
package com.example.amps_sdk.data

import com.example.amps_sdk.manager.AMPSEventManager

object InstanceChannelHelper {
    const val KEY_INSTANCE_ID = "instanceId"
    const val KEY_DATA = "data"

    fun instanceId(args: Any?): String? {
        val map = args as? Map<*, *> ?: return null
        return map[KEY_INSTANCE_ID] as? String
    }

    fun data(args: Any?): Any? {
        val map = args as? Map<*, *> ?: return null
        return map[KEY_DATA]
    }

    fun payload(instanceId: String, data: Any? = null): Map<String, Any?> {
        return mapOf(
            KEY_INSTANCE_ID to instanceId,
            KEY_DATA to data,
        )
    }

    fun send(method: String, instanceId: String, data: Any? = null) {
        AMPSEventManager.getInstance().sendMessageToFlutter(method, payload(instanceId, data))
    }
}

File: ios/Classes/common/InstanceChannelHelper.swift
//
//  InstanceChannelHelper.swift
//  amps_sdk
//
//  Flutter ↔ iOS 多实例协议辅助：{ instanceId, data }
//

import Foundation

enum InstanceChannelKeys {
    static let instanceId = "instanceId"
    static let data = "data"
}

enum InstanceChannelHelper {
    static func instanceId(from arguments: Any?) -> String? {
        guard let map = arguments as? [String: Any] else { return nil }
        return map[InstanceChannelKeys.instanceId] as? String
    }

    static func data(from arguments: Any?) -> Any? {
        guard let map = arguments as? [String: Any] else { return nil }
        return map[InstanceChannelKeys.data]
    }

    static func payload(instanceId: String, data: Any? = nil) -> [String: Any?] {
        return [
            InstanceChannelKeys.instanceId: instanceId,
            InstanceChannelKeys.data: data,
        ]
    }

    static func send(_ method: String, instanceId: String, data: Any? = nil) {
        AMPSEventManager.shared.sendToFlutter(method, arg: payload(instanceId: instanceId, data: data))
    }
}
